libkbfs/prefetcher_test.go

```diff
@@ -289,12 +291,30 @@ func TestPrefetcherDirectDirBlockAlreadyCached(t *testing.T) {
 	<-q.Prefetcher().Shutdown()
 
 	testPrefetcherCheckGet(
-		t, config.BlockCache(), dir2.Children["b"].BlockPointer, file1, false,
+		t, cache, dir2.Children["b"].BlockPointer, file1, false,
 		TransientEntry)
 	// Check that the dir block is marked as having been prefetched.
 	testPrefetcherCheckGet(
-		t, config.BlockCache(), dir1.Children["a"].BlockPointer, dir2, true,
+		t, cache, dir1.Children["a"].BlockPointer, dir2, true,
 		TransientEntry)
+
+	t.Log("Remove the prefetched file block from the cache.")
+	cache.DeleteTransient(dir2.Children["b"].BlockPointer, kmd.TlfID())
+	_, err = cache.Get(dir2.Children["b"].BlockPointer)
+	require.EqualError(t, err, NoSuchBlockError{dir2.Children["b"].BlockPointer.ID}.Error())
+
+	t.Log("Restart the prefetcher.")
+	q.TogglePrefetcher(context.Background(), true)
+
+	t.Log("Request the second-level directory block again. No prefetches should be triggered.")
```

Comment:
Is there a way to verify this?  Maybe try to get `dir2.Children["b"].BlockPointer` from the cache again after shutting down the prefetcher?